src/JuliaFormatter.jl

```diff
@@ -87,6 +93,17 @@ function Document(text::AbstractString)
                 # last comment
                 comments[sl] = (ws, cs)
             end
+
+            # There should not be more than 1 
+            # "off" tag on the stack at a time.
+            if t.val == "#! format: off" && length(stack) == 0
```

Comment:
Maybe a little more flexibility here would be good? A line matching `^#!\s*format\s*:\s*(on|off)\s*$`? One could potentially reformat the command line itself to be exactly `#! format: off` or `#! format: on` but I worry that it would be too easy to accidentally have trailing spaces or something and then be confused at why it doesn't work.

Reply:
makes sense, I'm just not too familiar with regexes and this was super simple :).

I guess `startswith(...)` could work too but then people could write whatever after the comment which could be weird. Might not actually be an issue in practice though.

Reply:
```suggestion
            if occursin(r"^#!\s*format\s*:\s*off\s*$", t.val) && length(stack) == 0
```

Reply:
```suggestion
            if occursin(r"^#!\s*format\s*:\s*off\s*$", t.val) && length(stack) == 0
```